gfx-memory/src/heaps/mod.rs

```diff
@@ -229,8 +235,14 @@ impl<B: hal::Backend> Heaps<B> {
 
 impl<B: hal::Backend> Drop for Heaps<B> {
     fn drop(&mut self) {
-        if !self.types.is_empty() {
-            log::error!("Heaps still have {:?} types live on drop", self.types.len());
+        for memory_heap in &self.heaps {
```

Comment:
What happens if the user doesn't force call `clear` but still free all the allocated blocks? We'd have them stored in `unused_blocks`, which means "not completely freed" still. I think we should support this case smoothly.
Perhaps, this `drop` implementation could go and at least force the freeing of the unused blocks?

Reply:
Previously the user was expected to manually call `Heaps::free`, I'm not sure why it was designed like that instead of just letting `Heaps::drop` handle it, regardless I tried to keep that design. (Maybe you have an idea why?)
However making `Heaps::drop` handle cleanup makes sense to me, i'll change it to do that.

Reply:
We should also remove the call to `Heaps::free` during shutdown in wgpu.

Reply:
Oh! its because it needs a device.

Reply:
Its not possible to have `Heaps::drop` handle the final cleanup as it needs access to the device..
In fact this wasnt even possible before as the cleanup that occured when freeing a single `LinearBlock` would always leave at least one line allocated.